ls1intum/Artemis
# Feature: Clean up participation and restart tour

### Checklist

### Motivation and Context
This PR should solve the issue https://github.com/ls1intum/Artemis/issues/1062 

### Description
When a user restarts a guided tours, in which the user had to participate in an exercise, then the `StudentParticipation`, as well as the `GuidedTourSetting` of the current tutorial will be deleted. The user will be directed to the exercise overview page of the the current course and can start the exercise. Once the user navigates to the exercise page, the tutorial will be automatically displayed to the user again.

Note : 
This PR adds the logic of deleting participations for guided tours, allowing the groups `USER` and `TA` to delete their own participations. (`/api/guided-tour/participations/{participationId}`)

### Steps for Testing
1. Log in to Artemis
2. Navigate to Tutorial Course
3. Visit guided tour for Modeling or Programming Exercise
4. Complete the tutorial and restart it
5. The `StudentParticipation` should have been deleted, when restarting the exercise, the tutorial should be displayed automatically from the beginning

### Screenshots
![restart](https://user-images.githubusercontent.com/33764166/72115583-df48b480-3347-11ea-8586-a59c8867d822.gif)
